cpp/src/parquet/column_writer.cc

```diff
@@ -460,7 +460,11 @@ class SerializedPageWriter : public PageWriter {
         ToThrift(page.definition_level_encoding()));
     data_page_header.__set_repetition_level_encoding(
         ToThrift(page.repetition_level_encoding()));
-    data_page_header.__set_statistics(ToThrift(page.statistics()));
+
+    // Write page statistics only when page index is not enabled.
+    if (column_index_builder_ == nullptr) {
+      data_page_header.__set_statistics(ToThrift(page.statistics()));
+    }
```

Comment:
This only disables writing the statistics, but still computes them, which is suboptimal.

Reply:
The stats computation is still required to build page index (a.k.a. the page stats of ColumnIndex).

Reply:
Ok, I see. Also, why do you use the wording "page index"? It's called column index in the spec.

Reply:
Hmm, that's true, my bad.